File: src/main/java/main/dto/request/UserRequest.java
package main.dto.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;
import main.dto.enums.RolesEnum;
import jakarta.validation.constraints.NotNull;
import org.springframework.validation.annotation.Validated;

@Validated
@Data
public class UserRequest{
    @NotNull
    @NotBlank
    private String name;

    @NotNull
    @NotBlank
    private String surname;

    @NotNull
    @NotBlank
    @Email(message = "Wrong format of email")
    private String email;

    @NotNull
    @NotBlank
    @Size(min = 8, message = "Password must be at least 8 symbols")
    private String password;

    @NotNull
    @NotBlank
    @Pattern(regexp = "^\\d{7,18}$", message = "Your phone number must contains 7-18 digits")
    private String phoneNumber;

    @NotNull(message = "Role is required")
    private RolesEnum role;
}

File: src/main/java/main/dto/UserRequest.kt
package main.dto

import jakarta.validation.constraints.AssertFalse
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Size
import org.jetbrains.annotations.NotNull
import org.springframework.validation.annotation.Validated

@Validated
data class UserRequest(
    @field:NotNull
    val name: String,
    @field:NotNull
    @field:Email(message = "Wrong format of email")
    val email: String,
    @field:NotNull
    @field:Size(min=8, message = "Password must be at least 8 characters long")
    val password: String,
    @field:NotNull
    @field:Size(min = 10)
    val phoneNumber: String
)
